### PR TITLE
[NAE-1668] PDF Generator 2022 Summer Update

### PR DESCRIPTION
**Pull request by renczesstefan**
# Description

Update PDF generator to support following funcionalities: grid layout together with TaskRefs; I18nField, UserList field, and divider; font customisation.

Implements [NAE-1668]

## Dependencies

No new dependencies were introduced

### Third party dependencies

No new dependencies were introduced

### Blocking Pull requests

There are no dependencies on other PR

## How Has Been This Tested?

This was tested manually generating PDFs.

### Test Configuration

| Name                | Tested on |
|---------------------| --------- |
| OS                  |   macOS Monterey 12.4        |
| Runtime             |  Java 11        |
| Dependency Manager  |  Maven 3.8.4         |
| Framework version   |  Spring Boot 2.6.2        |
| Run parameters      |           |
| Other configuration |           |

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes have been checked, personally or remotely, with @mladoniczky 
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have resolved all conflicts with the target branch of the PR
- [x] I have updated and synced my code with the target branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes:
    - [x] Lint test
    - [x] Unit tests
    - [x] Integration tests
- [x] I have checked my contribution with code analysis tools:
    - [x] [SonarCloud](https://sonarcloud.io/project/overview?id=netgrif_application-engine)
    - [x] [Snyk](https://app.snyk.io/org/netgrif/project/ea82b3b8-658d-41f5-89a7-76cd600da01f)
- [x] I have made corresponding changes to the documentation:
    - [x] Developer documentation
    - [x] User Guides
    - [x] Migration Guides

[NAE-1668]: https://netgrif.atlassian.net/browse/NAE-1668?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/com/netgrif/application/engine/pdf/generator/service/renderer/I18nDividerFieldRenderer.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    @Override
+    public void renderValue(PdfField field, int lineCounter) throws IOException {
+        setFieldParams(field);
+        renderValue(helperField, resource.getLabelFont(), resource.getFontGroupSize(), colorDataGroupLabel);
+    }
+
+    private void renderValue(PdfField field, PDType0Font font, int fontSize, Color colorLabel) throws IOException {
+        float textWidth = getTextWidth(Collections.singletonList(field.getLabel()), font, fontSize, resource);
+        int maxLineSize = getMaxLabelLineSize(field.getWidth(), fontSize);
+        List<String> multiLineText = new ArrayList<>() {{
```
Comment (minop) [inline src/main/java/com/netgrif/application/engine/pdf/generator/service/renderer/I18nDividerFieldRenderer.java:35]: bug marked by sonar.
If `field.getValues()` returns a `Collection`, then it should be possible to just call `new ArrayList<>(field.getValues())` to achieve the same result

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -123,9 +123,9 @@ server.error.include-stacktrace=never
 nae.pdf.resources.outputFolder=storage
 nae.pdf.resources.outputDefaultName=generated_pdf.pdf
 nae.pdf.resources.templateResource=file:src/main/resources/pdfGenerator/template_pdf.pdf
-nae.pdf.resources.font-title-resource=file:src/main/resources/pdfGenerator/fonts/Roboto-Bold.ttf
-nae.pdf.resources.font-label-resource=file:src/main/resources/pdfGenerator/fonts/Roboto-Bold.ttf
-nae.pdf.resources.font-value-resource=file:src/main/resources/pdfGenerator/fonts/Roboto-Bold.ttf
+nae.pdf.resources.font-title-resource=file:src/main/resources/pdfGenerator/fonts/Roboto-Light.ttf
```
Comment (minop) [inline src/main/resources/application.properties:126]: is this just a change of the default font? Or should the bold variant not be supported at all? Judging by the fact the light font .ttf file was added in this PR, then the bold .ttf file should be removed from the project (assuming it is a part of it).

**Review by minop — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties (hunk 2 above)
Comment (renczesstefan) [reply src/main/resources/application.properties:126]: This is a default font, the bold can be used as well, I can remove the unused fonts if you would like to.

File: src/main/java/com/netgrif/application/engine/pdf/generator/service/renderer/I18nDividerFieldRenderer.java (hunk 1 above)
Comment (renczesstefan) [reply src/main/java/com/netgrif/application/engine/pdf/generator/service/renderer/I18nDividerFieldRenderer.java]: corrected

File: src/main/resources/application.properties (hunk 2 above)
Comment (minop) [reply src/main/resources/application.properties:126]: if the bold font is never referenced then it should be removed in my opinion

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -122,7 +122,7 @@ server.error.include-stacktrace=never
 # PDF generator configuration
 nae.pdf.resources.outputFolder=storage
 nae.pdf.resources.outputDefaultName=generated_pdf.pdf
-nae.pdf.resources.templateResource=file:src/main/resources/pdfGenerator/template_pdf.pdf
+nae.pdf.resources.templateResource=classpath:pdfGenerator/template.pdf
 nae.pdf.resources.font-title-resource=file:src/main/resources/pdfGenerator/fonts/Roboto-Light.ttf
 nae.pdf.resources.font-label-resource=file:src/main/resources/pdfGenerator/fonts/Roboto-Light.ttf
```
Comment (minop) [inline src/main/resources/application.properties:127]: should the font files not also be referenced with `classpath:` and not via `file:`?
why were the light font files removed, if they are being referenced?

Comment (renczesstefan) [reply src/main/resources/application.properties]: Modified to classpath, and Roboto-light was not removed.

**Review by minop — APPROVED** [review]

**Review by mladoniczky — DISMISSED** [review]

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -122,10 +122,10 @@ server.error.include-stacktrace=never
 # PDF generator configuration
 nae.pdf.resources.outputFolder=storage
 nae.pdf.resources.outputDefaultName=generated_pdf.pdf
-nae.pdf.resources.templateResource=file:src/main/resources/pdfGenerator/template_pdf.pdf
-nae.pdf.resources.font-title-resource=file:src/main/resources/pdfGenerator/fonts/Roboto-Bold.ttf
-nae.pdf.resources.font-label-resource=file:src/main/resources/pdfGenerator/fonts/Roboto-Bold.ttf
-nae.pdf.resources.font-value-resource=file:src/main/resources/pdfGenerator/fonts/Roboto-Bold.ttf
+nae.pdf.resources.templateResource=classpath:pdfGenerator/template.pdf
```
Comment (machacjozef) [inline src/main/resources/application.properties:125]: file:.....

**Review by machacjozef — CHANGES_REQUESTED** [review]

**Review by machacjozef — APPROVED** [review]

**Review by mladoniczky — APPROVED** [review]

**Pull request merged**